experimental/sys.go

```diff
@@ -0,0 +1,13 @@
+package experimental
+
+// SysKey is a context.Context Value key. Its associated value should be a Sys.
+type SysKey struct{}
+
+// Sys controls experimental aspects currently only used by WASI.
+type Sys interface {
```

Comment:
ps naming.. ugh. We'll likely have more sysconfig at some point. Also, I don't want to necessarily say this is only for WASI. Since this is internal, we don't also need to force this name to be great, yet, but ideas welcome.